pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -100,22 +100,18 @@ func (mixerplugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.Mu
 		"context.reporter.kind": attrStringValue("outbound"),
 	}
 
-	switch in.ListenerProtocol {
-	case plugin.ListenerProtocolHTTP:
-		filter := buildOutboundHTTPFilter(in.Env.Mesh, attrs, in.Node)
-		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, filter)
-		}
-		return nil
-	case plugin.ListenerProtocolTCP:
-		filter := buildOutboundTCPFilter(in.Env.Mesh, attrs, in.Node, in.Service, in.Push)
-		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].TCP = append(mutable.FilterChains[cnum].TCP, filter)
+	httpFilter := buildOutboundHTTPFilter(in.Env.Mesh, attrs, in.Node)
+	tcpFilter := buildOutboundTCPFilter(in.Env.Mesh, attrs, in.Node, in.Service, in.Push)
```

Comment:
I don't think we should be building both filters unless they are used?

Reply:
is that a heavy operation? if not, its just a matter of preparing the necessary variables and adding it only when necessary

Reply:
Yes, it is. We did optimizations to avoid protobuf operations without a good reason. This is just wasteful to create proto and then discard it.

Reply:
This to me looks like it will double the load of mixerfilter. Every listener will be breaking creating both TCP and HTTP configs while before it was one of them.